cloud-storage.go

```diff
@@ -41,48 +41,61 @@ func NewCloudStorage(
 	gcpCredentialsJSON string,
 	gcpStorageEmulatorHost string,
 ) (CloudStorage, error) {
+	return NewCloudStorageWithOption(ctx, isTesting, bucketProvider, bucketName, CloudStorageOption{
+		AWSS3Endpoint:          awsS3Endpoint,
+		AWSS3Region:            awsS3Region,
+		AWSS3AccessKeyID:       awsS3AccessKeyID,
+		AWSS3SecretAccessKey:   awsS3SecretAccessKey,
+		AWSEnableS3Accelerate:  false,
+		GCPCredentialsJSON:     gcpCredentialsJSON,
+		GCPStorageEmulatorHost: gcpStorageEmulatorHost,
+	})
+}
+
+//nolint:funlen
+func NewCloudStorageWithOption(ctx context.Context, isTesting bool, bucketProvider, bucketName string, cloudStorageOpts CloudStorageOption) (CloudStorage, error) {
 	switch bucketProvider {
 	case "", "aws":
 		// 3-rd party library uses global variables
-		if awsS3AccessKeyID != "" {
-			err := os.Setenv("AWS_ACCESS_KEY_ID", awsS3AccessKeyID)
+		if cloudStorageOpts.AWSS3AccessKeyID != "" {
+			err := os.Setenv("AWS_ACCESS_KEY_ID", cloudStorageOpts.AWSS3AccessKeyID)
 			if err != nil {
 				return nil, err
 			}
 		}
 
 		// 3-rd party library uses global variables
-		if awsS3SecretAccessKey != "" {
-			err := os.Setenv("AWS_SECRET_ACCESS_KEY", awsS3SecretAccessKey)
+		if cloudStorageOpts.AWSS3SecretAccessKey != "" {
+			err := os.Setenv("AWS_SECRET_ACCESS_KEY", cloudStorageOpts.AWSS3SecretAccessKey)
 			if err != nil {
 				return nil, err
 			}
 		}
 
 		if isTesting {
-			return newAWSTestCloudStorage(ctx, awsS3Endpoint, awsS3Region, bucketName)
+			return newAWSTestCloudStorage(ctx, cloudStorageOpts.AWSS3Endpoint, cloudStorageOpts.AWSS3Region, bucketProvider)
 		}
 
-		return newAWSCloudStorage(ctx, awsS3Endpoint, awsS3Region, bucketName)
+		return newAWSCloudStorage(ctx, cloudStorageOpts.AWSS3Endpoint, cloudStorageOpts.AWSS3Region, bucketProvider, &cloudStorageOpts.AWSEnableS3Accelerate)
```

Comment:
This seems producing a bug in `0.4.0` because `bucketProvider` is being used instead of `bucketName`. Could someone clarify? Thanks!